dds/idl/dds_generator.h

```diff
@@ -940,7 +1025,18 @@ class Fields {
   }
 
 private:
-  AST_Structure* node_;
+  AST_Structure* const node_;
+  const bool explicit_keys_only_;
 };
 
+inline
+size_t array_element_count(AST_Array* arr)
+{
+  size_t count = 1;
+  for (size_t i = 0; i < arr->n_dims(); ++i) {
```

Comment:
To avoid warnings, match the return type (and `count`, `i`) with the type of `n_dims` and `ulval`, which are both 32-bit unsigned integers.